### PR TITLE
Add note about clone in docs for vec![]

### PR DESCRIPTION
**Pull request by Manishearth**
r? @steveklabnik

### CODE REVIEW

File: src/libcollections/macros.rs [hunk 1]
```diff
@@ -32,6 +32,12 @@
 /// Note that unlike array expressions this syntax supports all elements
 /// which implement `Clone` and the number of elements doesn't have to be
 /// a constant.
+///
+/// This will use `clone()` to duplicate an expression, so one should be careful
+/// using this with types having a nonstandard `Clone` implementation. For
+/// example, `vec![Rc::new(1), 5]` will create a vector of five references
+/// to the same boxed integer value, not five references pointing to independent
+/// boxed integer.
```
Comment (steveklabnik) [inline src/libcollections/macros.rs]: This should be

> pointing to independently boxed integers

**Comment by steveklabnik** [comment]
r=me after nit

**Comment by Manishearth** [comment]
@bors r=steveklabnik rollup

agh, was going to write "independent integers" and forgot to change it properly

File: src/libcollections/macros.rs [hunk 2]
```diff
@@ -32,6 +32,12 @@
 /// Note that unlike array expressions this syntax supports all elements
 /// which implement `Clone` and the number of elements doesn't have to be
 /// a constant.
+///
+/// This will use `clone()` to duplicate an expression, so one should be careful
+/// using this with types having a nonstandard `Clone` implementation. For
+/// example, `vec![Rc::new(1), 5]` will create a vector of five references
```
Comment (Gankro) [inline src/libcollections/macros.rs]: Rc::new(1); 5

File: src/libcollections/macros.rs [hunk 3]
```diff
@@ -32,6 +32,12 @@
 /// Note that unlike array expressions this syntax supports all elements
 /// which implement `Clone` and the number of elements doesn't have to be
 /// a constant.
+///
+/// This will use `clone()` to duplicate an expression, so one should be careful
+/// using this with types having a nonstandard `Clone` implementation. For
+/// example, `vec![Rc::new(1) 5]` will create a vector of five references
```
Comment (Gankro) [inline src/libcollections/macros.rs]: Still wrong :(

**Comment by Manishearth** [comment]
@bors r=steveklabnik rollup

agh, fixed

**Pull request merged**